### PR TITLE
Feature style settings

### PR DESCRIPTION
**Pull request by holroy**
This complements #2 to allow changing of various style attributes to be changed through the _Style Settings_ plugin if that's installed. If it's not installed, all values will retain their default values. 

If one would like to, one could of course extend which variables are to be set using the style settings option.

( The commit includes the compilation fixes which I also made PR for related to  #23)

### CODE REVIEW

File: src/styles.css [hunk 1]
```diff
@@ -1,3 +1,117 @@
...
+    default: '#242424'
+*/
+:root {
+  --ht21-cell-fontfamily: "Source Code Pro", monospace;
+	--ht21-cell-font-size: 13px;
+	--ht21-border-width: 1px;
+	--ht21-border-color: #363636;
+	--ht21-tick-height: 16px;
+	--ht21-tick-width: 20px;
+	--ht21-tick-color: #8a5cf5;
```
Comment (zoreet) [inline src/styles.css:108]: This should be assigned `var(--checkbox-color)` otherwise it will break HT24 in themes other than the default obsidian theme. the same applies to all the cases where a variable was used before: --ht21-border-width, --ht21-row-hover-color. Will that still work with the plugin you're trying to support?

Comment (holroy) [reply src/styles.css:108]: I'm not sure how to use a variable in this type of setting.  Let me ponder a little on that, and feel free to suggest a solution if you see one. After all this is your project,  and I just wanted to chip in on how to make it even better and more customisable.

**Comment by holroy** [comment]
Now I've reset the variables to match as closely as I could the default values and variables of Obsidian. Not entirely sure if the defaults are the same in light and dark mode. And after going a little bananas and changing the colors my habit tracker looked like this:
image

And even though the colors are hard-coded, I could reset and it would fall back and follow the variables, and it then looks like this:

image

---

In other words, if you leave it as default it'll follow the definition of the variabels within the chosen color scheme, and if you change any particular color that specific color (or setting) will be changed. And if you later on want to reset to default for either one or all settings it'll allow that as well.

**Comment by zincplusplus** [comment]
@holroy after almost 2 years I got back to OSS and released v2 which has extensive css variables to configure everything https://github.com/zincplusplus/habit-tracker/blob/master/styles.css

Are you still using this plugin/obsidian and would like to give it a try? I'm curios to hear what you think and if there's stuff I've missed
